collect_app/src/main/java/org/odk/collect/android/activities/InstanceUploaderActivity.java

```diff
@@ -93,14 +96,42 @@ protected void onCreate(Bundle savedInstanceState) {
         long[] selectedInstanceIDs = null;
         if (savedInstanceState != null && savedInstanceState.containsKey(TO_SEND)) {
             selectedInstanceIDs = savedInstanceState.getLongArray(TO_SEND);
+
+            // Add optional authentication params
+            if (savedInstanceState.containsKey(ApplicationConstants.BundleKeys.URL) && savedInstanceState.containsKey(ApplicationConstants.BundleKeys.USERNAME) && savedInstanceState.containsKey(ApplicationConstants.BundleKeys.PASSWORD)) {
```

Comment:
Please split into 3 lines to help with readability.

Reply:
> How does this work with servers that allow anonymous access? Does passing in the URL and empty strings for username and password work?

Yes, it works if the username & password are empty strings
I tested with https://opendatakit.appspot.com but it seems to accept any username/password. Any other server that you would recommend me to test with?

Reply:
Awesome. That's the only special case I can think of!